docs/src/main/sphinx/functions/table.rst

```diff
@@ -69,3 +69,52 @@ You cannot mix the argument conventions in one invocation.
 
 All arguments must be constant expressions, and they can be of any SQL type,
 which is compatible with the declared argument type.
+
+.. _table-functions-query:
+
+The ``query`` function
```

Comment:
This is not a generic function. It's a connector-specific function that happens to be named "query" in various connectors.

Reply:
I’m not sure the third one makes sense in the long term. For instance, we’re about to add a “query” function to the Elasticsearch connector that uses completely different syntax and semantics. It would be wrong to try jam it into the documentation for the query function for other connectors.

Also, even for the “query” function for jdbc-based connectors there will be some differences regarding what syntax can be used (database-specific), type mappings, etc., and the examples we might want to highlight.

Reply:
My thought is that you'd explain the function and how to generically invoke it in Trino, and leave the examples to the connectors. But perhaps that's best done with a fragment and going more with the first idea, so there's not some weird standalone page. I'll put up an edit with that approach once I've written an example or two for every connector.

Reply:
@martint @colebow I suggest the following approach:

1. we document the `query` function per connector. For each connector, we add this description:
```
(The `query` function) allows you to query
the underlying database directly. The query function requires syntax native to
the connector, because the full query is pushed down and performed in-place.
This can be useful for accessing native features which are not implemented in
Trino, or improving query performance where running a query natively may be
faster.
```
 Whether we copy-paste it, or use a fragment, is an implementation detail.
After the description, we add a couple connector-specific examples.

2. In `table.rst`, we list all the Table Functions with a short description and a link to the connector page. We could categorize them by connectors, so that we could add more functions in the future in a natural way. It would be something like:
```
Druid
- `query` table function for full query pass-through.

MariaDB
- `query` table function for full query pass-through.
```

3. We can consider adding a dedicated docs page for "query pass-through". There we could gather the info of what's currently supported (plus links), what are the differences etc. We could send people to that page whenever they ask for query pass-through instead of confronting them with the theory of Table Functions.